onnxruntime/python/onnxruntime_inference_collection.py

```diff
@@ -346,6 +346,13 @@ def _create_inference_session(self, providers, provider_options, disabled_optimi
                                                                         provider_options,
                                                                         available_providers)
 
+        if providers == [] and len(available_providers) > 1:
+            warnings.warn("Deprecation warning. This ORT build has {} enabled.".format(available_providers) +
```

Comment:
looks like the warn api has a way to specify it as a DeprecationWarning.
i.e.
warnings.warn("deprecation warning" , warnings.DeprecationWarning)

Reply:
I tested adding DeprecationWarning before, but it can't show warning message when running Python.

Reply:
https://docs.python.org/3/library/exceptions.html#DeprecationWarning
It seems ignored by the default warning filters.

Reply:
ok i see. that's too bad.